### PR TITLE
win_domain: Add debug log

### PR DESCRIPTION
**Pull request by agowa**
##### SUMMARY

Add a debug log.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
win_domain

##### ADDITIONAL INFORMATION

No functional change, I just added some debug logging statements.

### CODE REVIEW

File: lib/ansible/modules/windows/win_domain.py [hunk 1]
```diff
@@ -71,6 +71,11 @@
     type: str
     choices: [ Win2003, Win2008, Win2008R2, Win2012, Win2012R2, WinThreshold ]
     version_added: '2.8'
+  log_path:
```
Comment (ShachafGoldstein) [inline lib/ansible/modules/windows/win_domain.py:74]: The indentations seem off

Comment (agowa) [reply lib/ansible/modules/windows/win_domain.py:74]: Are you a bot? You're always so fast with reviewing ;-)
Is now fixed.

**Comment by jhawkesworth** [comment]
Nice idea but wondering whether it might be better to convert the module to use the new Ansible.Basic csharp module util and then you can log to the windows event log.

**Comment by agowa338** [comment]
jhawkesworth: Why windows event log? Should on linux we also don't write ansible module logs into syslog, or is that the new 'it'?

**Comment by agowa338** [comment]
> I think it is safe to say that none of the 'win_' modules are going to be useful on linux. Even if they were modified to work with latest powershell running under .net core, it would be better to use the existing linux modules.

That's not what I was referring to. I was referring to the linux modules on linux. They all provide debug output on the controller. So that if you debug your playbook you get the output instead of having to ssh to the remote host and look into syslog.

**Comment by jhawkesworth** [comment]
Sorry for the confusion.  Since the PR is would add ability to write log files to the target host, not the controller, I was just questioning whether having module-specific code to handle logging on the target host could be avoided by using shared module utils code.

**Comment by agowa338** [comment]
rerun ci

**Comment by agowa338** [comment]
ready_for_review

**Comment by agowa338** [comment]
ready_for_review

**Comment by agowa338** [comment]
!needs_revision

**Comment by agowa338** [comment]
bot_status

**Comment by agowa338** [comment]
bot_status

**Comment by jborean93** [comment]
Sorry for the ring around here @agowa338, while other modules do have something similar to this it's not really a use case that I want to encourage in other modules. If it's for testing of a module then I recommend you actually debug the module when running it locally as that is a lot easier to do and identify what is actually happening.

This basically reflects what @jhawkesworth has stated. On Windows the event viewer is used as that is the common place where events are logged on the host. There's not a real need for this to be replicated to a flat file. If you want to add this functionality to this module you will have to first convert it to the `Ansible.Basic` style and take advantage of the log methods there.

**Comment by agowa338** [comment]
I'm fine with that, just wanted to share the code, as I've used it to debug the module.

**Pull request closed without merging**